WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java

```diff
@@ -137,53 +90,81 @@ public void finish() {
     @Override
     protected void onResume() {
         super.onResume();
+        showCorrectToolbar();
         mIsRunning = true;
         ActivityId.trackLastActivity(ActivityId.THEMES);
 
-        // fetch themes if we don't have any
-        if (NetworkUtils.isNetworkAvailable(this) && WordPress.getCurrentBlog() != null
-                && WordPress.wpDB.getThemeCount(getBlogId()) == 0) {
-            fetchThemes(mViewPager.getCurrentItem());
-            setRefreshing(true, mViewPager.getCurrentItem());
-        }
+        fetchThemesIfNoneAvailable();
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        getSupportFragmentManager().popBackStack();
```

Comment:
Can you clarify why this is necessary? Also, the rest of the Activity uses `getFragmentManager()` but this is using `getSupportFragmentManager()`. Any specific reason?